Nucleus terraform update to pull platform linux/amd64 compatible docker images on any operating system

ramesh-maddegoda opened the pull request:
## 🗒️ Summary
Earlier it was required to execute Terraform of Nucleus from an EC2 instance with linux/amd64 architecture to make sure the docker images pulled and used in ECS tasks are compatible with  linux/amd64 architecture. This pull request makes an update that allows pulling and pushing platform linux/amd64 compatible docker images from any platform (this allows to use Nucleus terraform on platforms other than linux/amd64).

## ⚙️ Test Data and/or Report
Terraform Nucleus was tested on a MacBook with Apple M2 Pro chip. Before this change the Validate and Harvest Airflow tasks failed (as shown in the workflow below in red color) because it used docker images incompatible with linux/amd64 used in ECS tasks. 

After making this change, it was possible to execute Nucleus Terraform from same platform (which is not  linux/amd64 ) and it was able to pull docker images compatible with linux/amd64 used in ECS tasks successfully. As a result the workflow shown in the following screenshot was passed (shown in green color).

[
<img width="323" height="360" alt="Screenshot 2025-08-05 at 12 26 07 AM" src="https://github.com/user-attachments/assets/9490ead9-cb15-4280-bc75-71e49da3d0cb" />
](url)

## ♻️ Related Issues
Deployment improvement make Nucleus Terraform platform independent

nutjob4life reviewed (approved):
Code deletion is almost always automatic approval 🎉 

As for secrets, does the baseline need to be updated?

ramesh-maddegoda commented:
> Code deletion is almost always automatic approval 🎉
> 
> As for secrets, does the baseline need to be updated?

Thanks @nutjob4life ! I did a secret detection and updated the secret baseline anyway for the safe side to avoid any mistakes (specially since I updated a template file that deals with account IDs etc.).
